services/src/main/java/org/exoplatform/addons/gamification/listener/social/space/GamificationSpaceListener.java

```diff
@@ -144,30 +142,31 @@ public void spaceBannerEdited(SpaceLifeCycleEvent event) {
 
   private void createGamificationHistoryEntry(String username, Space space, String ruleTitle) {
     // Compute user id
-    String senderId = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, username).getId();
-    String receiverId = identityManager.getOrCreateIdentity(SpaceIdentityProvider.NAME, space.getPrettyName()).getId();
-    String spaceURL = "/portal/g/:spaces:" + space.getGroupId().replace("/spaces/", "");
+    String senderId = identityManager.getOrCreateUserIdentity(username).getId();
+    String receiverId = identityManager.getOrCreateSpaceIdentity(space.getPrettyName()).getId();
 
     try {
       Map<String, String> gam = new HashMap<>();
       gam.put("ruleTitle", ruleTitle);
-      gam.put("object", spaceURL);
+      gam.put("objectId", receiverId);
+      gam.put("objectType", IDENTITY_OBJECT_TYPE);
       gam.put("senderId", senderId);
       gam.put("receiverId", receiverId);
       listenerService.broadcast(GENERIC_EVENT_NAME, gam, null);
     } catch (Exception e) {
-      LOG.error("Cannot broadcast gamification event");
+      LOG.error(CANNOT_BROADCAST_GAMIFICATION_EVENT);
```

Comment:
Please fix this old bug by avoiding exception swallow (same for other parts):

```suggestion
      LOG.error("Error while broadcasting gamification event: {}", gam, e);
```